hooks/pre-commit

```diff
@@ -1,25 +1,27 @@
-#!/bin/bash
-if clang-format --version | grep -q 'version 11\.'; then
-   CLANG_FORMAT_EXECUTABLE="clang-format"
-else
-   CLANG_FORMAT_EXECUTABLE="clang-format-11"
-fi
+#!/bin/sh
+
+name=""
+version="14.0.0"
 
-if ! command -v $CLANG_FORMAT_EXECUTABLE &> /dev/null
-then
-    echo $CLANG_FORMAT_EXECUTABLE does not exist, make sure to install it
-    exit 1
+for file in $(find $(echo "$PATH" | sed 's/:/ /g') -maxdepth 1 -type f -executable -name 'git-clang-format*'); do
```

Comment:
ShellCheck also complains about this line. You should take a look at that.

Reply:
> The reason it needs to be a for loop rather than a while read loop or find -execdir is because it needs to modify the name and version variables.

Are you sure that's a problem? Shell is pretty lenient about accessing variables.

Reply:
A while read loop could work, but the only way to do it in a POSIX-compliant way is by sending the output from `find` to a temporary file, then reading that file for the loop. This is because if you use the standard pipe from `find` to the while loop, the loop becomes a subshell, and variables inside won't be available outside.

Do you think that that's a good solution?